tests/conftest.py

```diff
@@ -30,6 +31,11 @@ def pytest_runtest_teardown(item, nextitem):
         marks = [m.name for m in item.own_markers ]
         if 'parallel' not in marks:
             pyccel_clean(path_dir, remove_shared_libs = True)
+        else:
+            comm = MPI.COMM_WORLD
+            comm.Barrier()
+            if comm.rank == root:
+                pyccel_clean(path_dir, remove_shared_libs = True)
```

Comment:
You may need to add another barrier after cleaning

Reply:
Fixed in ec0cc27